pkg/loop/internal/pb/relayer.proto

```diff
@@ -0,0 +1,306 @@
+  rpc NewMedianProvider (NewMedianProviderRequest) returns (NewMedianProviderReply) {}
+  rpc NewMercuryProvider (NewMercuryProviderRequest) returns (NewMercuryProviderReply) {}
+
+  rpc ChainStatus (ChainStatusRequest) returns (ChainStatusReply) {}
+  rpc ChainStatuses (ChainStatusesRequest) returns (ChainStatusesReply) {}
+
+  rpc NodeStatuses (NodeStatusesRequest) returns (NodeStatusesReply) {}
+
+  rpc SendTx (SendTxRequest) returns (google.protobuf.Empty) {}
+}
+
+// RelayArgs represents [github.com/smartcontractkit/chainlink-relay/pkg/types.RelayArgs].
+message RelayArgs {
+  bytes externalJobID = 1; // [32]byte
+  int32 jobID = 2;
+  string contractID = 3;
+  bool new = 4;
+  bytes relayConfig = 5;
+}
+
+// RelayArgs represents [github.com/smartcontractkit/chainlink-relay/pkg/types.PluginArgs].
+message PluginArgs {
+  string transmitterID = 1;
+  bytes pluginConfig = 2;
+}
+
+// NewConfigProviderRequest has arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.NewConfigProvider].
+message NewConfigProviderRequest {
+  RelayArgs relayArgs = 1;
+}
+
+// NewConfigProviderReply has return arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.NewConfigProvider].
+message NewConfigProviderReply {
+  uint32 configProviderID = 1;
+}
+
+// NewMedianProviderRequest has arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.NewMedianProvider].
+message NewMedianProviderRequest {
+  RelayArgs relayArgs = 1;
+  PluginArgs pluginArgs = 2;
+}
+
+// NewMedianProviderReply has return arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.NewMedianProvider].
+message NewMedianProviderReply {
+  uint32 medianProviderID = 1;
+}
+
+// NewMercuryProviderRequest has arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.NewMercuryProvider].
+message NewMercuryProviderRequest {
+  RelayArgs relayArgs = 1;
+  PluginArgs pluginArgs = 2;
+}
+
+// NewMercuryProviderReply has return arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.NewMercuryProvider].
+message NewMercuryProviderReply {
+  uint32 mercuryProviderID = 1;
+}
+
+// ChainStatusRequest has arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.ChainStatus].
+message ChainStatusRequest {
+  string id = 1;
+}
+
+// ChainStatusReply has return arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.ChainStatus].
+message ChainStatusReply {
+  ChainStatus chain = 1;
+}
+
+// ChainStatusesRequest has arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.ChainStatuses].
+message ChainStatusesRequest {
+  int32 offset = 1;
+  int32 limit = 2;
+}
+
+// ChainStatusesReply has return arguments for [github.com/smartcontractkit/chainlink-relay/pkg/loop.Relayer.ChainStatuses].
+message ChainStatusesReply {
+  repeated ChainStatus chains = 1;
+  int32 count = 2; // total count of chains
```

Comment:
Doesn't the above repeated field already support a length() argument?
Why also return count?

Reply:
These responses are paged - `chains` may only be a subset based on the `offset` and `length` args that were used.

Reply:
I mean, if offset = 5, limit is 10 in request.
And the node has say 20 chains configured.

So reply will have this?
chains has length 10(chain ids 5 to 14).
count = 20

Is this what the response will be?